### ISSUE TITLE
Cryptic errors thrown for illegal use of `R.chain`

### ISSUE THREAD

**Issue by dmitriz** [ramda/ramda#2529]
I've got this error when writing [this answer](https://stackoverflow.com/questions/45786580/ramda-chain-usage/50345438#50345438):

```js
// Define our own uncurried append
const appendCustom = (a, b) => R.append(a, b)
R.chain(appendCustom, R.head)([1, 2]);
```
It is almost the same as the doc example,
except the append function is intentionally uncurried.
The [REPL throws the cryptic error message](http://ramdajs.com/repl/?v=0.25.0#?%2F%2F%20Define%20our%20own%20uncurried%20append%0Aconst%20appendCustom%20%3D%20%28a%2C%20b%29%20%3D%3E%20R.append%28a%2C%20b%29%0AR.chain%28appendCustom%2C%20R.head%29%28%5B1%2C%202%5D%29%3B%0A):

```sh
t(...) is not a function
```
which is possibly a bug?
The usage is of course illegal but on a somewhat subtle level,
ignored by the incorrect `R.chain` implementation 
written in the top voted example to this question.

**Comment by CrossEye** [comment]
Yes, Ramda does have something of a problem with cryptic error messages.  I just don't know a good way around this which doesn't end up doing something like Sanctuary's expensive run-time type-checking.

But of course, this is an incorrect application.

```js
appendCustom('foo') //=> ['foo']
```

... which might not be great, but it's the result of calling `appendCustom('foo', undefined)`.  The point is that this is not a function.

Clearly we could fix this by actual currying:

```js
const appendCustom = (a) => (b) => R.append(a, b)
```

or by Ramda-style currying:

```js
const appendCustom = R.append
```

But I don't think there is anything to do here.

**Comment by dmitriz** [comment]
What makes it cryptic is the reference to 
```js
t(...) is not a function
```
Because there is no such function facing the user.

The JS regular
```js
undefined is not a function
```
while not great, would be already more helpful,
because now the user knows to look for undefined values attempted to call as functions.

This would be even better (in fact, excellent):
```js
appendCustom(...) is not a function
```
Now as user I know exactly what is the problem.

Is there perhaps any magic that can be undone to achieve that? ;)
In fact, this is how this case is already handled in Node:
```sh
> adding = (x, y) => x+y
[Function: adding]
> adding(1)(2)
TypeError: adding(...) is not a function
```

**Comment by dmitriz** [comment]
Just came across this post, could that be the missing sauce?
https://medium.com/@drboolean/debugging-functional-7deb4688a08c

**Comment by CrossEye** [comment]
@dmitriz: Yes, I think that is the secret sauce, and I want to include it post 1.0.

**Comment by dmitriz** [comment]
Awesome! 😄

**Comment by iamstarkov** [comment]
I wish 1.0 would come sooner
